elytron/WFCORE-4483-microprofile-jwt.adoc

```diff
@@ -0,0 +1,73 @@
+= WFCORE-4483 Add support for missing MP-JWT requirements
+:author:            Martin Mazanek
+:email:             mmazanek@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+There are two main security pieces that are currently missing but will be needed for MP JWT support whenever that happens:
+
+* Support for injection of the currently authenticated caller as a JsonWebToken
+* The java.security.Principal returned from javax.ws.rs.core.SecurityContext.getUserPrincipal() must be an instance of org.eclipse.microprofile.jwt.JsonWebToken
+
+This issue focuses on adding these requirements.
+
+https://www.eclipse.org/community/eclipse_newsletter/2017/september/article2.php[Eclipse MP-JWT overview]
+
+This issue is not intended as an official support for MP-JWT, so this is not targeted towards specific MP-JWT spec version. Development will be done using version 1.1.1.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/EAP7-1259[EAP7-1259]
+* https://issues.jboss.org/browse/WFCORE-4483[WFCORE-4483]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/ELY-1834[ELY-1834]
+* https://issues.jboss.org/browse/WFLY-1834[WFLY-1834]
+
+=== Dev Contacts
+
+* mailto:mmazanek@redhat.com[Martin Mazanek]
+
+=== QE Contacts
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+[x] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+* WildFly, Security
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
```

Comment:
The requirements make sense to me, but they are a bit vague. Could you be a bit more specific about the code in which the change is going to be made?